### ISSUE TITLE
JsonApiDocument must not have data and errors field?

### ISSUE THREAD

**Issue by dopin** [michiel/jsonapi-rust#23]
Hi, thanks for creating a nice crate!

I noticed that when I create a json formatted string like following code, it includes `errors` with a `null` value which must not included with `data` on the top level.
The code is a part of a server side JSON API which returns a data successfully.

```rust
let a = JsonApiDocument {
  data: Some(PrimaryData::Multiple(resources)),
  errors: None,
  meta: None,
  included: None,
  links: None,
  jsonapi: None,
}

let string = serde_json::to_sring(&a).unwrap();
```

According to the specification of JSONAPI, it says:

>The members data and errors MUST NOT coexist in the same document.

http://jsonapi.org/format/#document-top-level

and also, its [json.schema](http://jsonapi.org/faq/#is-there-a-json-schema-describing-json-api) is defined to not allow to have `errors`  for success documents.
I thought that it's probably ok to have `errors` with `null` but it looks no.

Here's [my quick fix](https://github.com/dopin/jsonapi-rust/commit/71e65a8822235e359029c32af51a23bc911fb37d) to remove `errors` and `data` keys if it's a `None`.
But I think that the code can be more suitable by making `JsonApiDocument` into an enum and divide it according in three, success, failure and info.

**Comment by michiel** [comment]
Hi @dopin, thanks!

You are correct. The `errors` **field** should not be included with `data` and the other way around. Can you make a pull request of your change?

I like your idea of making `JsonApiDocument` into an enum, it feels cleaner and would mean less code. I will make it an issue and try it out.

**Comment by dopin** [comment]
Thanks for reading.
I've created a PR. https://github.com/michiel/jsonapi-rust/issues/24 😳 

I should have called `errors` `errors` *field*. I learned English. 📝  😄

**Comment by michiel** [comment]
I've merged the PR, thank you!

**Comment by michiel** [comment]
Published v0.5.1 fyi

**Comment by dopin** [comment]
Thank you! 😀
